### ISSUE TITLE
java.lang.IllegalArgumentException - Ticking World

### ISSUE THREAD

**Issue by Laifsyn** [MrFantiVideo/MineGate#27]
https://bytebin.lucko.me/xjhqmHIeSK

```---- Minecraft Crash Report ----
// Ouch. That hurt :(

Time: 5/22/22, 12:26 AM
Description: Exception ticking world

java.lang.IllegalArgumentException: Cannot set property class_2754{name=type, clazz=class net.minegate.fr.moreblocks.block.enums.FernType, values=[default, plant]} as it does not exist in Block{ecologics:prickly_pear}
	at Not Enough Crashes deobfuscated stack trace.(1.18.2+build.3)
	at net.minecraft.state.State.with(State:117)
	at net.minecraft.block.CropBlock.handler$fkd002$randomTick(CropBlock:3111)
	at net.minecraft.block.CropBlock.randomTick(CropBlock:91)
	at samebutdifferent.ecologics.block.PricklyPearBlock.randomTick(PricklyPearBlock.java:51)
	at net.minecraft.block.AbstractBlock$AbstractBlockState.randomTick(AbstractBlock:923)
	at net.minecraft.server.world.ServerWorld.redirect$fcb000$redirectBlockStateTick(ServerWorld:13171)
	at net.minecraft.server.world.ServerWorld.tickChunk(ServerWorld:486)
	at net.minecraft.server.world.ServerChunkManager.tickChunks(ServerChunkManager:378)
	at net.minecraft.server.world.ServerChunkManager.tick(ServerChunkManager:322)
	at net.minecraft.server.world.ServerWorld.tick(ServerWorld:311)
	at net.minecraft.server.MinecraftServer.redirect$cim000$fixUpdateSuppressionCrashTick(MinecraftServer:5790)
	at net.minecraft.server.MinecraftServer.tickWorlds(MinecraftServer:907)
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer:851)
	at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer:97)
	at net.minecraft.server.MinecraftServer.handler$cbi000$modifiedRunLoop(MinecraftServer:4896)
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer:676)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer:273)
	at java.lang.Thread.run(Unknown Source)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.class_2688.method_11657(class_2688.java:117)
	at net.minecraft.class_2302.handler$fkd002$randomTick(class_2302.java:3111)
	at net.minecraft.class_2302.method_9514(class_2302.java:91)
	at samebutdifferent.ecologics.block.PricklyPearBlock.method_9514(PricklyPearBlock.java:51)
	at net.minecraft.class_4970$class_4971.method_26199(class_4970.java:923)
	at net.minecraft.class_3218.redirect$fcb000$redirectBlockStateTick(class_3218.java:13171)
	at net.minecraft.class_3218.method_18203(class_3218.java:486)
	at net.minecraft.class_3215.method_14161(class_3215.java:378)
	at net.minecraft.class_3215.method_12127(class_3215.java:322)
	at net.minecraft.class_3218.method_18765(class_3218.java:311)
	at net.minecraft.server.MinecraftServer.redirect$cim000$fixUpdateSuppressionCrashTick(MinecraftServer.java:5790)

-- Affected level --
Details:
	All players: 1 total; [class_3222['Laifsyn'/499, l='ServerLevel[New World]', x=216.12, y=64.00, z=297.93]]
	Chunk stats: 2401
	Level dimension: minecraft:overworld
	Level spawn location: World: (208,65,304), Section: (at 0,1,0 in 13,4,19; chunk contains blocks 208,-64,304 to 223,319,319), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-64,0 to 511,319,511)
	Level time: 523 game time, 523 day time
	Level name: New World
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: true
	Level weather: Rain time: 137513 (now: false), thunder time: 178510 (now: false)
	Known server brands: fabric
	Level was modded: true
	Level storage version: 0x04ABD - Anvil
Stacktrace:
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:907)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:851)
	at net.minecraft.class_1132.method_3748(class_1132.java:97)
	at net.minecraft.server.MinecraftServer.handler$cbi000$modifiedRunLoop(MinecraftServer.java:4896)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:676)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:273)
	at java.base/java.lang.Thread.run(Unknown Source)

-- System Details --
Details:
	Minecraft Version: 1.18.2
	Minecraft Version ID: 1.18.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.1, Eclipse Adoptium
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Eclipse Adoptium
	Memory: 155890824 bytes (148 MiB) / 4831838208 bytes (4608 MiB) up to 4831838208 bytes (4608 MiB)
	CPUs: 12
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i7-5820K CPU @ 3.30GHz
	Identifier: Intel64 Family 6 Model 63 Stepping 2
	Microarchitecture: unknown
	Frequency (GHz): 3.30
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: NVIDIA GeForce GTX 970
	Graphics card #0 vendor: NVIDIA (0x10de)
	Graphics card #0 VRAM (MB): 4095.00
	Graphics card #0 deviceId: 0x13c2
	Graphics card #0 versionInfo: DriverVersion=30.0.15.1123
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 2.13
	Memory slot #0 type: DDR4
	Virtual memory max (MB): 21425.51
	Virtual memory used (MB): 15388.17
	Swap memory total (MB): 13312.00
	Swap memory used (MB): 1537.68
	JVM Flags: 5 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx4608m -Xms4608m -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xms256m
	Fabric Mods: 
		additionaladditions 4.0.2
		additionalstructures 3.0.3
		adorn 3.5.0+1.18.2
		advancements-enlarger 0.2.4
		adventurez 1.4.10
		ae2 11.0.0
			team_reborn_energy 2.0.0-beta1
		ae2things 1.1.0-beta.5
		ae2wtlib 11.0.1
		alloy_forgery 2.0.13+1.18
		allstackable 1.5.4-1.18.1
		alternate-current 1.2.1
		ambientenvironment 7.0+19
		antighost 1.18.2-fabric0.47.10-1.1.4
			crowdin-translate 1.4+1.18.2
		appleskin 2.4.0+mc1.18.2
		architectury 4.4.60
		autoconfig1u 3.4.0
		automated_crafting 1.4.4
		awesomedungeon 2.0.8
		awesomedungeonend 1.0.1
		awesomedungeonnether 1.0.1
		awesomedungeonocean 2.0.6
		backpackmod 1.18.2-3.1.1
		backslot 1.2.6
			amecsapi 1.3.5+mc22w06a
		backslotaddon 1.0.8
		badpackets 0.1.2
		balm-fabric 3.0.0+0
		basicshields 1.3.1-1.18.2
		bclib 1.4.6
		beaconoverhaul 1.6.0+1.18.2
			reach-entity-attributes 2.1.1
		better_withered_mobs 1.7.0
		betterdungeons 1.18.2-Fabric-2.1.0
		bettereating 0.3.0
		betterend 1.1.0
		betterf3 1.2.5
		betterfpsdist 1.18-1.7
		bettermineshafts 1.18.2-Fabric-2.1.3
		betternether 6.1.1
		betterstrongholds 1.18.2-Fabric-2.1.1
		betterthanmending 1.3.0
		bettertridentreturn 1.18.2-1.0.2
		biomemakeover 1.18.2-1.4.21
		blockus 2.4.5+1.18.2
			terraform-wood-api-v1 3.0.0
		blur 2.4.1
			midnightlib 0.3.1
			satin 1.7.0
		bobby 3.1.1
			com_typesafe_config 1.4.2
			io_leangen_geantyref_geantyref 1.3.13
			org_spongepowered_configurate-core 4.1.2
			org_spongepowered_configurate-hocon 4.1.2
		bookshelf 13.0.13
		bottledair-fabric 1.0
		breakprogress 1.0.1
		buildguide 0.3.2
		byg 1.4
		c2me 0.2.0+alpha.7.0
			com_ishland_c2me_libs_ 0.2.0+alpha.7.0
		cameraoverhaul 1.3.0-fabric-universal
		can-i-mine-this-block 1.2.0
		carpet 1.4.69
		cem 0.7.1
			completeconfig 1.0.0
		cenchants 1.13.0
		chalk 2.0.1+1.18.2
		charcoalplus 1.1.2
		chunky 1.2.164
		clearview 4.0.1
		clickadv 2.6
		clickthrough 1.18.2-fabric0.47.10-0.4
			gbfabrictools 1.3.4+1.18.2
		clienttweaks 7.1.0
		cloth-config 6.2.57
			cloth-basic-math 0.6.0
		clumps 8.0.0+8
		collective-fabric 4.26
		companion 2.0.1
		concreteconversion 6.1.0.0
		connectiblechains 2.1.1+1.18.2
		controlling 9.0+19
		cosmetic-armor 1.3.1
		couplings 1.8.0+1.18.2
			com_electronwill_night-config_core 3.6.5
			com_electronwill_night-config_toml 3.6.5
		crafttweaker 9.1.140
			org_javassist_javassist 3.28.0-GA
			org_reflections_reflections 0.10.2
		crawl 0.10.6
			mm 2.3
		creativecore 2.6.10
		creeperspores 1.8.0
		crops_love_rain 1.2.2
		culinaire 2.4.0
		cullleaves 2.3.2
		damagetilt 1.18-fabric-0.1.2
		dark-enchanting 1.0.0+1.18.2
			libgui 5.4.0+1.18.2
				jankson 4.0.0+j1.2.0
					blue_endless_jankson 1.2.1
				libninepatch 1.1.0
			omega-config 1.2.3-1.18.1
		dawn 3.3.0
		deathlog 0.2.8+1.18
		dehydration 1.2.8
		doublejump 1.0.1+mc1.18
		dragonloot 1.0.10
		dynamicfps 2.1.0
			com_moandjiezana_toml_toml4j 0.7.2
		ecologics 1.5.4
		edenring 0.5.1
		effective 1.2.1
		enchant_the_rainbow 1.0.1
		enchdesc 10.0.3
		enhancedvisuals 1.5.4
		essential_commands 0.23.0-mc1.18.2
			com_fasterxml_jackson_dataformat_jackson-dataformat-yaml 2.13.0
			ec-core 1.0.1-mc1.18.2
			fabric-permissions-api-v0 0.1-SNAPSHOT
			placeholder-api 1.1.3+1.17.1
		expert_armor 1.7.5
		extended_armor_bars 1.2.1
		extrasponges 1.3.0
		fabric 0.53.0+1.18.2
			fabric-api-base 0.4.4+d7c144a860
			fabric-api-lookup-api-v1 1.6.0+2373a54560
			fabric-biome-api-v1 8.0.4+7b0f312660
			fabric-blockrenderlayer-v1 1.1.12+3ac43d9560
			fabric-command-api-v1 1.1.9+d7c144a860
			fabric-commands-v0 0.2.8+b4f4f6cd60
			fabric-containers-v0 0.1.21+d7c144a860
			fabric-content-registries-v0 3.0.4+d82b939260
			fabric-convention-tags-v1 1.0.1+474e4f7960
			fabric-crash-report-info-v1 0.2.0+a6b0947860
			fabric-dimensions-v1 2.1.17+2540745460
			fabric-entity-events-v1 1.4.8+d7c144a860
			fabric-events-interaction-v0 0.4.19+d7c144a860
			fabric-events-lifecycle-v0 0.2.15+d7c144a860
			fabric-game-rule-api-v1 1.0.13+d7c144a860
			fabric-item-api-v1 1.5.0+85719ab760
			fabric-item-groups-v0 0.3.12+91896a4960
			fabric-key-binding-api-v1 1.0.12+54e5b2ec60
			fabric-keybindings-v0 0.2.10+b4f4f6cd60
			fabric-lifecycle-events-v1 2.0.3+2d83e92c60
			fabric-loot-tables-v1 1.0.11+d7c144a860
			fabric-mining-level-api-v1 2.1.1+81ea5b3c60
			fabric-models-v0 0.3.6+d7c144a860
			fabric-networking-api-v1 1.0.21+d882b91560
			fabric-networking-v0 0.3.8+b4f4f6cd60
			fabric-object-builder-api-v1 2.1.1+032c981d60
			fabric-particles-v1 0.2.12+526dc1ac60
			fabric-registry-sync-v0 0.9.8+0d9ab37260
			fabric-renderer-api-v1 0.4.13+d882b91560
			fabric-renderer-indigo 0.5.0+7faf0d8860
			fabric-renderer-registries-v1 3.2.11+b4f4f6cd60
			fabric-rendering-data-attachment-v1 0.3.7+d7c144a860
			fabric-rendering-fluids-v1 2.0.2+54e5b2ec60
			fabric-rendering-v0 1.1.13+b4f4f6cd60
			fabric-rendering-v1 1.10.7+54e5b2ec60
			fabric-resource-conditions-api-v1 2.0.3+91b7aa6660
			fabric-resource-loader-v0 0.5.0+2afceb2960
			fabric-screen-api-v1 1.0.10+d882b91560
			fabric-screen-handler-api-v1 1.2.0+1f6558e860
			fabric-textures-v0 1.0.12+3ac43d9560
			fabric-transfer-api-v1 1.6.1+f4563ac860
			fabric-transitive-access-wideners-v1 1.0.0+bd8a4a1860
		fabric-language-kotlin 1.7.4+kotlin.1.6.21
			org_jetbrains_kotlin_kotlin-reflect 1.6.21
			org_jetbrains_kotlin_kotlin-stdlib 1.6.21
			org_jetbrains_kotlin_kotlin-stdlib-jdk7 1.6.21
			org_jetbrains_kotlin_kotlin-stdlib-jdk8 1.6.21
			org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.6.1
			org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.6.1
			org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm 1.3.2
			org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.3.2
			org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.3.2
		fabriccrossdimtpfix 0.3.0
		fabricloader 0.14.4
		fabricshieldlib 1.5.3-1.18
		fabrishot ${version}
		fallingtree 3.5.1
		farmersdelight 1.18.2-0.2.1
		feature_nbt_deadlock_be_gone 2.0.0+1.18.2-rc1
		ferritecore 4.2.1
		fireplacelib 6.3.1+1.18.2
			annotateddi 2.1.0+1.18.2
				aopalliance_aopalliance 1.0
				com_google_inject_guice 5.1.0
				javax_inject_javax_inject 1
		flan 1.18.2-1.7.8
		forgeconfigapiport 3.2.0
		fst 2.3.2+1.18.2
		geckolib3 3.0.37
			com_eliotlash_mclib_mclib 18
			com_eliotlash_molang_molang 18
		geodes 1.4.2
		globalspawn 1.5.2-1.18.2
		goldenhoppers 1.3.0
		harvest_scythes 2.5.0
		health_levels 1.0.2
		healthcare 1.0.11
			config2brigadier 1.0.2
		healthoverlay 6.0.1
			fiber 0.23.0-2
		heartbond 1.3.0
		herdspanic 1.0.5
		horseinfo 0.4.1-1.18
		horsestatsvanilla 4.1.10
		improvedmobs 1.18.2-1.10.9
		incantationem 1.1.3+1.18.2
		incapacitated 1.1.2-1.18
		indium 1.0.3+mc1.18.2
		ingredient-extension-api 2.0.5
		inventoryprofilesnext 1.3.8
		iris 1.2.4-build.38
			org_anarres_jcpp 1.4.14
		itemscroller 0.16.0
		java 17
		kibe 1.9.7-BETA+1.18
			playerabilitylib 1.5.0
		kiwi 6.0.2
		krypton 0.1.9
			com_velocitypowered_velocity-native 1.1.0-SNAPSHOT
		ksyxis 1.1
		lambdynlights 2.1.0+1.17
			pride 1.1.0+1.17
			spruceui 3.3.0+1.17
				org_aperlambda_lambdajcommon 1.8.1
		lazydfu 0.1.2
		lengthyladders 1.0.5
		libraryferret 1.0.3
		lightoverlay 6.0.5
		limitedchunks 1.18.2-2.5
		litematica 0.11.2
		lithium 0.7.9
		macawsbridgesbyg 1.18.2-1.2
		macawsroofsbyg 1.18.2-1.0
		magnumtorch 3.1.2
		malilib 0.12.1
		mcwbridges 2.0.2
		mcwdoors 1.0.6
		mcwfences 1.0.4
		mcwlights 1.0.3
		mcwroofs 2.1.2
		mcwtrpdoors 1.0.5
		mcwwindows 2.0.3
		megane 7.5.1
			megane-api 7.1.0
			megane-applied-energistics-2 7.1.0
			megane-fabric-transfer 7.1.0
			megane-kibe 7.1.0
			megane-reborn-energy 7.0.0
			megane-runtime 7.2.0
			megane-vanilla 7.1.1
		midashunger 2.1.0+1.18.2
		minecarttweaks 1.5
		minecraft 1.18.2
		minegate-moreblocks 1.2.0
		minieffects 4.1.0
		minihud 0.22.0
		modmenu 3.2.2
		morevillagers-fabric 3.2.0-SNAPSHOT
		mostructures 1.4.2+1.18.2
		mousetweaks 2.22
		multiworldborders 1.4-fabric-1.18.2
		mythicmetals 0.13.1
			magna 1.7.0-1.18-pre1
		mythicmetals_decorations 0.2.1
			ninjaphenix_container_lib 1.3.0
		naturescompass 1.18.2-2.0.3-fabric
		netherchest 3.1
		netherportalfix 9.0.0
		nomoreglowingpots 1.0.2
		notenoughcrashes 4.1.6+1.18.2
		nyfsquiver Fabric-0.5.0-1.18.2
		onsoulfire 1.18-3
			cardinal-components-base 4.1.4
			cardinal-components-entity 4.1.4
		overlord 3.1.5+1.18.2
		owo 0.7.2+1.18
		oxidized 1.6.4
		patchouli 1.18.2-67-FABRIC
		paxi 1.18.2-Fabric-2.0
		pehkui 3.3.1+1.14.4-1.19
			kanos_config 0.3.1+1.14.4-1.19
		piglib 1.1.1
		polymorph 0.0.21-1.18.2
			cardinal-components-block 4.1.4
			cardinal-components-item 4.1.4
		preciseblockplacing 1.8
		presencefootsteps 1.5.0
			kirin 1.10.0
		promenade 2.6.1
			terraform-shapes-api-v1 2.1.0
		puzzleslib 3.3.1
		quartzelv 2.1.1+1.18
		rainbegoneritual-fabric 1.7
		recipecache 0.2.1-1.17.1
		reeses-sodium-options 1.4.2
		repurposed_structures 4.1.8+1.18.2
		reroll 1.4.0-1.18.1
			omega-config-gui 1.2.1-1.18.1
				cloth-config-lite 2.0.6
		resounding 1.0.0-bc.9
		rightclickharvest 2.0.3
		roughlyenoughitems 8.1.457
		roughlyenoughresources 2.5.0
		rpgdifficulty 1.0.11
		scaffoldingdropsnearby-fabric 1.5
		secretrooms 1.3.1+1.18.2
			advanced_runtime_resource_pack 0.5.7
		servercore 1.2.9-1.18.2
		shulkerboxtooltip 3.0.7+1.18.2
		smallernetherportals-fabric 1.8
		sodium 0.4.1+build.15
			org_joml_joml 1.10.2
		sodium-extra 0.4.4+mc1.18.2-build.39
			caffeineconfig 1.1.0
		spark 1.9.1
		starlight 1.0.2+fabric.89b8d9f
		starterkit-fabric 3.1
		stoneholm 1.4.3
		sulfurpotassiummod 1.1.2
		tempad 1.2.1
		tenshilib 1.18.2-1.6.7
		terrablender 1.1.0.99
		throwabletorch 1.18.2-1
		toolstats 9.0.1
		tooltipfix 1.0.4-1.18
		torchbowmod 1.0
		totw_additions 1.2.1
		tradingpost 3.2.0
		trampledisablerfabric 1.0.4
		transparent_cosmetics 1.4.0
		transportables 1.2
		trinkets 3.3.1
		tweakeroo 0.13.3
		universal_ores 1.2.1
		veggie_way 1.2.32
			pugh_lib 1.3.0
		vehiclefix 3.3
		waystones 2.5.3
		wmch 1.4.4
		wmitaf 2.2.0
		wolveswitharmor 1.7.1-1.18.2
			maybe-data 1.2.1-1.18.2
			static-content 1.0.1-1.16.2
				staticdata 1.1.2
		worldedit 7.2.10+1742f98
		wthit 4.10.3
		yigd 1.3.5
		yungsapi 1.18.2-Fabric-2.0.8
		yungsbridges 1.18.2-Fabric-2.1.0
		yungsextras 1.18.2-Fabric-2.1.0
	Loaded Shaderpack: (off)
	Server Running: true
	Player Count: 1 / 8; [class_3222['Laifsyn'/499, l='ServerLevel[New World]', x=216.12, y=64.00, z=297.93]]
	Data Packs: vanilla, Fabric Mods, minecraft/eamodcompat
	World Generation: Stable
	Type: Integrated Server (map_client.txt)
	Is Modded: Definitely; Client brand changed to 'fabric'; Server brand changed to 'fabric'
	Client Crashes Since Restart: 0
	Integrated Server Crashes Since Restart: 2
	Suspected Mods: Ecologics (ecologics)```

**Comment by Laifsyn** [comment]
I do have generate Prickly Pears as true though

**Comment by MrFantiVideo** [comment]
This is the same issue as #26 related to the new version.
It will be fixed in the next version which also fixes the problem #5.

The option « useMixins » will no longer be necessary and generation problems with other mods will be over.

You can temporarily use this version: [MoreBlocks 1.2.1-dev-2](https://libraries.minegate.fr/maven/net/minegate/fr/minegate-moreblocks/1.2.1-dev2/minegate-moreblocks-1.2.1-dev-2.jar)

**Comment by MrFantiVideo** [comment]
Fixed in the new version. ([MoreBlocks 1.2.1](https://github.com/MrFantiVideo/MineGate/releases/tag/1.2.1))
